config/regex.js

```diff
@@ -41,7 +41,8 @@ module.exports = {
     // Steps can only be named with A-Z,a-z,0-9,-,_
     STEP_NAME: /^[\w-]+$/,
     // Jobs can only be named with A-Z,a-z,0-9,-,_
-    JOB_NAME: /^[\w-]+$/,
+    // Also allow stage setup/teardown like stage@stage-name:setup
+    JOB_NAME: /^(([\w-]+)|(stage@[\w-]+:(setup|teardown)))$/,
```

Comment:
wondering if something like this would help with matching/grouping 
```suggestion
    JOB_NAME: /^(([\w-]+)|(?:stage@([\w-]+):(setup|teardown)))$/,
```
<img width="374" alt="Screenshot 2023-10-27 at 3 23 57 PM" src="https://github.com/screwdriver-cd/data-schema/assets/3230529/7e16757a-a056-4bfb-8ab6-5c4eebb4dc6f">